### PR TITLE
[SuperTextField] Fix selection glitch when using IME (Resolves #905)

### LINKED ISSUE #905: Selection glitch in SuperTextField when IME is used
The selection in `SuperTextField` jumps between old and current selection as shown in this recording:

https://user-images.githubusercontent.com/31278849/210235669-7d2d08b7-414b-4aed-b087-8c5e071b2013.mov

This is reproducible in demo app ("SuperTextField" -> "Interactive demo"). If `TextInputSource.keyboard` is used instead, then the issue is not reproducible.

### PR DESCRIPTION
**Pull request by angelosilvestre**
[SuperTextField] Fix selection glitch when using IME. Resolves #905

When using `SuperTextField` and typing a character, the selection jumps between the old and the new selected position. When using `SuperDesktopTextField` this isn't happening.

`SuperTextField` is always using the `defaultTextFieldKeyboardHandlers` by default. Because of this, typing is being handled by `insertCharacterWhenKeyIsPressed` instead of being handled using deltas.

This PR changes `SuperTextField` to use `defaultTextFieldImeKeyboardHandlers` when in IME mode, the same way `SuperDesktopTextField` does. With this change, the issue stopped.

### CODE REVIEW

**Comment by angelosilvestre** [comment]
This caused some tests to fail. I will need to investigate.

**Comment by angelosilvestre** [comment]
The `SuperTextField` initialization had to be modified.

File: super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart [hunk 1]
```diff
@@ -334,6 +319,19 @@ class SuperTextFieldState extends State<SuperTextField> {
   }
 }
 
+TextInputSource platformDefaultInputSource() {
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart:322]: Is there a reason that this was moved? I don't see any functional differences.

File: super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart [hunk 2]
```diff
@@ -64,10 +64,15 @@ class SuperTextField extends StatefulWidget {
...
     this.maxLines = 1,
     this.lineHeight,
-    this.inputSource,
-    this.keyboardHandlers = defaultTextFieldKeyboardHandlers,
+    TextInputSource? inputSource,
+    List<TextFieldKeyboardHandler>? keyboardHandlers,
     this.padding,
-  }) : super(key: key);
+  })  : inputSource = inputSource ?? platformDefaultInputSource(),
+        keyboardHandlers = keyboardHandlers ??
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart:71]: Can you do this in a local method, like how we dealt with the input source before?

File: super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart (hunk 1 above)
Comment (angelosilvestre) [reply super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart:322]: We need this method to be outside of the state class, so we can use it in the `SuperTextField` initializer

File: super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart (hunk 2 above)
Comment (angelosilvestre) [reply super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart:71]: We can't access local methods in an initializer. Also, the methods are in the state class...

File: super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart (hunk 1 above)
Comment (matthew-carroll) [reply super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart:322]: I guess what I'm confused about is why we need to access this from the widget initializer. Isn't the widget initializer just making a choice about which input source and which handlers to use? Why can't we make that decision in the `State` object?

Comment (angelosilvestre) [reply super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart]: It seems we can make it a lot simpler. As `SuperDesktopTextField` already chooses its default handlers, we could just make the property nullable in `SuperTextField`.

**Review by matthew-carroll — APPROVED** [review]
LGTM - Please cherry pick to stable

**Comment by matthew-carroll** [comment]
Oops. @angelosilvestre looks like there are some conflicts, somehow

**Comment by angelosilvestre** [comment]
@matthew-carroll Conflicts solved

**Pull request merged**